### PR TITLE
Improve markdown conversion quality

### PR DESCRIPTION
**Pull request by GBKS**
Refactor HTML-to-Markdown converter to eliminate navigation chrome and noise artifacts, improving content quality for AI consumption.

Based on BOLTy's feedback [in this comment](https://github.com/BitcoinDesign/Guide/pull/1228#pullrequestreview-4147680789).

### CODE REVIEW

File: netlify/edge-functions/markdown-negotiation.ts [hunk 1]
```diff
@@ -1,6 +1,34 @@
...
+  ".side-bar",
+  ".secondary-nav",
+  ".search-input-wrap",
+  ".social-media-list",
+  ".footer-nav",
+  ".skip-to-content-link",
+  ".header-anchor",
+  ".next-previous",
+  ".next-previous--feedback",
+  ".home-banner-info",
```
[the remark this answers is not in the document]
Comment (GBKS) [reply netlify/edge-functions/markdown-negotiation.ts:25]: Copilot in Visual Studio Code disagrees with this, so I'll resolve.

File: netlify/edge-functions/markdown-negotiation.ts [hunk 2]
```diff
@@ -408,3 +453,70 @@ function toAbsoluteUrl(href: string | null, baseUrl: URL): string {
...
+function pruneDocument(doc: { querySelectorAll: (selector: string) => any }): void {
+  for (const selector of NOISE_SELECTORS) {
+    doc.querySelectorAll(selector).forEach((node: any) => node.remove());
+  }
+}
+
+function selectContentRoot(doc: { querySelector: (selector: string) => any; body?: any }): any {
+  const main = doc.querySelector("main");
+  if (main && /\bhome-page\b/.test(main.getAttribute?.("class") || "")) {
+    return main;
```
[the remark this answers is not in the document]
Comment (GBKS) [reply netlify/edge-functions/markdown-negotiation.ts:466]: Copilot in Visual Studio Code disagrees with this, so I'll resolve.

**Comment by swedishfrenchpress** [comment]
this is about over my head. i'm not confident i can give an accurate / valuable approval / review.

**Comment by GBKS** [comment]
@swedishfrenchpress all good. Interestingly, the best entity to give feedback on this PR are AI agents, since it's about improving how they can understand the site content.

**Comment by sbddesign** [comment]
> this is about over my head. i'm not confident i can give an accurate / valuable approval / review.

It's probably not over your agent's head

**Review by sbddesign — APPROVED** [review]
tl;dr - this is a huge improvement, though there are edge cases you could defend against

My first reaction was "LGTM".

I asked BOLTy, and he sees it as a clear improvement over the one before.

However, there are some edge cases where some of the stripping might be too aggressive. For example, code samples in inline SVG and pages that don't follow the structured article format.

I don't actually think the guide contains any of those three things, so I think you're good to go. However, if you wanted to be cautious in case such things existed in the guide in the future, you could feed both these feedback below into your coding agent to have it fixed. 

### BOLTy's Feedback

This is a clear improvement for agent-readability. I compared the deploy preview against live, including https://bitcoin.design/guide/daily-spending-wallet/, and the preview is much better at surfacing actual page content instead of nav/sidebar noise.

A few things worth feeding back to the agent before merge:

Fallback path may break HTML code samples
In the regex fallback flow, noise stripping happens before `<pre>` blocks are protected. If a guide page includes HTML examples with tags like `<nav>`, `<header>`, or `<footer>`, those examples could get mangled.

SVG stripping may be overly aggressive
Removing inline `<svg>` is probably fine for chrome/icons, but it could drop meaningful content if any pages use inline SVG diagrams or illustrations. Worth checking a few representative pages.

Content-root selection should be spot-checked on different page types
The new root-selection logic is directionally good, but it’s opinionated. I’d sanity-check homepage, guide article pages, and section/index pages to make sure it’s not narrowing too aggressively or skipping useful content.

Overall: strong improvement, just worth validating those edge cases.

**Comment by GBKS** [comment]
Thank you. I will merge this and do another local test to see if the script can be improved. If so, I'll create a new PR.
